### PR TITLE
performance boost and new props design

### PR DESCRIPTION
**Pull request by endurance21**
* not clearing whole canvas for animatin , only part taken by firefly is being cleared in each animationframe
 * made the library less flexible but more usefull 
 * only props we need now are 
   *  canvasWidth
   * canvasHeight
   * colors array
* other props has to be removed asap from documentation

### CODE REVIEW

**Comment by endurance21** [comment]
@Adi-11  kindly review the pr and we need to update the readme soon and also we may need to make the example more beautifull and eye catchy !

File: src/index.tsx [hunk 1]
```diff
@@ -28,15 +25,23 @@ const giveBirth :any = throttle(function (mouse,numberOfBirths,colors,changeDire
...
+} , 50)
 const setFromEvent = (e,setMousePosition,boundary) =>{
   setMousePosition({ x: e.clientX - boundary.current.x, y: e.clientY - boundary.current.y });
 }
-const ReactFirefly = (props)=>{
-  let {numberOfBirths, canvasWidth, canvasHeight,colors,changeDirectionFrequency,randomMotion} = props
-  if(numberOfBirths === null){
-    numberOfBirths = 2
-  }
+const ReactFirefly = (props:props)=>{
```
Comment (Adi-11) [inline src/index.tsx:32]: Better to use
```
interface ReactFireFlyPropTypes {
 canvasWidth:number ;
 canvasHeight:number;
 colors:string[]
}
export const DataGridDemo: React.FC<ReactFireFlyPropTypes> = ({
    canvasWidth, 
    canvasHeight, 
    colors 
 }) => { 
    .......
    return (<Canvas canvasRef={canvas} ></Canvas>)
} 
```
much more readable and clean!!! @endurance21

**Review by Adi-11 — COMMENTED** [review]
Will soon update the documentations

File: src/index.tsx (hunk 1 above)
Comment (endurance21) [reply src/index.tsx:32]: loved that idea ! you can make changes and comit to this pr?

**Comment by endurance21** [comment]
@Adi-11 alos can you check this working in you local env. i.e by makig a sample project ?

File: src/index.tsx (hunk 1 above)
Comment (Adi-11) [reply src/index.tsx:32]: Yup!! sure i will make all the changes and make necessary updates in doc. by the EOD.

**Comment by endurance21** [comment]
@Adi-11 i am in hurry, merging this one and you can create a separtae pr for the same

**Pull request merged**